src/ui/components/panels/settings/agent-settings/general-settings-form/general-settings-form.tsx

```diff
@@ -20,20 +20,51 @@ import {
 
 import "twin.macro";
 
-export const GeneralSettingsForm = () => (
-  <>
-    <div tw="w-[400px]">
-      <DisabledFormGroup fields={[{ name: "id", label: "AGENT ID" }, { name: "agentType", label: "AGENT TYPE" }]} />
-    </div>
-    <DarkFormGroup label="Agent name">
-      <Field name="name" component={Fields.DarkInput} placeholder="Enter agent's name" />
-    </DarkFormGroup>
-    <DarkFormGroup label="Description" optional>
-      <Field
-        name="description"
-        component={Fields.DarkTextarea}
-        placeholder="Add agent's description"
-      />
-    </DarkFormGroup>
-  </>
-);
+interface Props {
+  type: string;
+}
+
+interface labelsAndPlaceholdersData {
+  idLabel: string,
+  nameLabel: string,
+  namePlaceholder: string,
+  descriptionPlaceholder: string,
+}
+
+export const GeneralSettingsForm = ({ type }: Props) => {
+  const data: labelsAndPlaceholdersData = {
```

Comment:
```suggestion
  const labelsAndPlaceholdersData = {
```
Typescript will understand that this object has neccessary properties